lib/models/device.coffee

```diff
@@ -275,23 +275,20 @@ exports.identify = (uuid, callback) ->
 # @param {String} newName - the device new name
 # @param {module:resin.models.device~renameCallback} callback - callback
 #
-# @todo This action doesn't return any error
-# if trying to rename a device that does not
-# exists. This should be fixed server side.
-#
 # @example
 #	resin.models.device.rename 317, 'NewName', (error) ->
 #		throw error if error?
 #		console.log("Device has been renamed!")
 ###
 exports.rename = (name, newName, callback) ->
-	return pine.patch
-		resource: 'device'
-		body:
-			name: newName
-		options:
-			filter:
-				name: name
+	exports.get(name).then ->
```

Comment:
It's prohibited for apps in a hook https://bitbucket.org/rulemotion/resin-api/src/413cdc75c4d372dd23b719c4427da3da04de1f02/src/models/hooks/index.coffee?at=master#cl-39